src/app/core/market/api/bid/bid.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { MarketService } from 'app/core/market/market.service';
+import { MarketStateService } from 'app/core/market/market-state/market-state.service';
+import { SnackbarService } from 'app/core/snackbar/snackbar.service';
+
+import { Cart } from 'app/core/market/api/cart/cart.model';
+import { Listing } from 'app/core/market/api/listing/listing.model';
+
+@Injectable()
+export class BidService {
+
+  constructor(private market: MarketService,
+              private marketState: MarketStateService,
+              private snackbar: SnackbarService) {
+  }
+
+  order(cart: Cart, profile: any): Promise<boolean> {
+    let nBidsPlaced = 0;
+
+    return new Promise((resolve, reject) => {
+      cart.listings.forEach((listing: Listing) => {
+        if (listing.hash) {
+          // bid for item
+          this.market.call('bid', ['send', listing.hash, profile.address])
+            .subscribe(
+              (res) => {
+                if (++nBidsPlaced === cart.listings.length) {
+                  this.snackbar.open('Order has been successfully placed');
+                  resolve(true);
+                }
+              },
+              (failed) => {
+                reject(false);
```

Comment:
Does it notify you of failure? There are messages that come back from the bid code that tell you why it failed